spec/requests/works_spec.rb

```diff
@@ -311,6 +311,32 @@
       end
     end
 
+    describe 'update work with an attachment' do
+      let(:work) { create(:work, :with_attached_file) }
+      let(:user) { work.depositor }
+      let(:work_params) do
+        {
+          title: 'New title',
+          work_type: 'text',
+          contact_email: 'io@io.io',
+          abstract: 'test abstract',
+          attached_files_attributes: {
+            '0' => { 'label' => 'two', '_destroy' => '', 'hide' => '0', 'id' => work.attached_files.first.id }
+          },
+          keywords_attributes: {
+            '0' => { '_destroy' => 'false', 'label' => 'Feminism', 'uri' => 'http://id.worldcat.org/fast/922671' }
+          },
+          license: 'CC0-1.0',
+          release: 'immediate'
+        }
+      end
+
+      it 'redirects to the work page' do
+        patch "/works/#{work.id}", params: { work: work_params }
```

Comment:
It's not clear to me how this test tests the change above. What would the behavior of this test be w/o the change to a virtual attr?

Reply:
It raises the error outlined in #388